crates/rolldown/src/stages/generate_stage/code_splitting.rs

```diff
@@ -41,82 +41,48 @@ impl GenerateStage<'_> {
-      let count: u32 = entry_index.try_into().expect("Too many entries, u32 overflowed.");
-      let mut bits = BitSet::new(entries_len);
-      bits.set_bit(count);
-      let Module::Normal(module) = &self.link_output.module_table.modules[entry_point.id] else {
-        continue;
-      };
-      let chunk = chunk_graph.add_chunk(Chunk::new(
-        entry_point.name.clone(),
-        entry_point.reference_id.clone(),
-        entry_point.file_name.clone(),
-        bits.clone(),
-        vec![],
-        ChunkKind::EntryPoint {
-          is_user_defined: module.is_user_defined_entry,
-          bit: count,
-          module: entry_point.id,
-        },
-        self.link_output.lived_entry_points.contains(&entry_point.id),
-      ));
-      bits_to_chunk.insert(bits, chunk);
-      entry_module_to_entry_chunk.insert(entry_point.id, chunk);
-    }
 
-    // Determine which modules belong to which chunk. A module could belong to multiple chunks.
-    self.link_output.entries.iter().enumerate().for_each(|(i, entry_point)| {
-      if self.options.is_hmr_enabled() {
-        // If HMR is enabled, we need to make sure it belongs to at least one chunk even if no module reaches it.
-        self.determine_reachable_modules_for_entry(
-          self.link_output.runtime.id(),
-          i.try_into().expect("Too many entries, u32 overflowed."),
-          &mut index_splitting_info,
-        );
-      }
-      self.determine_reachable_modules_for_entry(
-        entry_point.id,
-        i.try_into().expect("Too many entries, u32 overflowed."),
-        &mut index_splitting_info,
-      );
-    });
+    if self.options.preserve_modules {
+      let modules_len = self
+        .link_output
+        .module_table
+        .modules
+        .len()
+        .try_into()
+        .expect("Entry length exceeds u32::MAX");
+      for (idx, module) in self.link_output.module_table.modules.iter_enumerated() {
+        let Module::Normal(module) = module else {
+          continue;
+        };
 
-    let mut module_to_assigned: IndexVec<ModuleIdx, bool> =
-      oxc_index::index_vec![false; self.link_output.module_table.modules.len()];
-
-    self.apply_advanced_chunks(&index_splitting_info, &mut module_to_assigned, &mut chunk_graph);
-
-    // 1. Assign modules to corresponding chunks
-    // 2. Create shared chunks to store modules that belong to multiple chunks.
-    for normal_module in self.link_output.module_table.modules.iter().filter_map(Module::as_normal)
-    {
-      if !normal_module.meta.is_included() {
-        continue;
+        let count = idx.raw();
+        let mut bits = BitSet::new(modules_len);
+        bits.set_bit(count);
+        let chunk = chunk_graph.add_chunk(Chunk::new(
+          None,
+          None,
+          None,
+          bits.clone(),
+          vec![],
+          ChunkKind::EntryPoint {
+            is_user_defined: module.is_user_defined_entry,
```

Comment:
Could you check out if rollup treats them as user-defined modules? Also if rollup try to split them in a entry chunk or common chunk level?

Reply:
Yeah, related code is here https://github.com/rollup/rollup/blob/061a0387c8654222620f602471d66afd3c582048/src/Chunk.ts?plain=1#L230-L232